ga4gh/datarepo.py

```diff
@@ -179,9 +200,457 @@ def __init__(
+            CREATE TABLE CallSet (
+                id TEXT,
+                variantSetId TEXT,
+                name TEXT
+            );
+        """
+        cursor.execute(sql)
+
+    def insertCallSet(self, callSet):
+        """
+        Inserts a the specified callSet into this repository.
+        """
+        sql = """
+            INSERT INTO CallSet (
+                id, variantSetId, name)
+            VALUES (?, ?, ?);
+        """
+        cursor = self._dbConnection.cursor()
+        cursor.execute(sql, (
+            callSet.getId(), callSet.getParentContainer().getId(),
+            callSet.getLocalId()))
+
+    def _readCallSetTable(self, cursor):
+        cursor.row_factory = sqlite3.Row
+        cursor.execute("SELECT * FROM CallSet;")
+        for row in cursor:
+            variantSet = self.getVariantSet(row[b'variantSetId'])
+            callSet = variants.CallSet(variantSet, row[b'name'])
+            callSet.populateFromRow(row)
+            assert callSet.getId() == row[b'id']
+            # Insert the callSet into the memory-based object model.
+            variantSet.addCallSet(callSet)
+
+    def _createVariantSetTable(self, cursor):
+        sql = """
+            CREATE TABLE VariantSet (
+                id TEXT,
+                datasetId TEXT,
+                referenceSetId TEXT,
+                name TEXT,
+                created TEXT,
+                updated TEXT,
+                metadata TEXT,
+                dataUrlIndexMap TEXT
+            );
+        """
+        cursor.execute(sql)
+
+    def insertVariantSet(self, variantSet):
+        """
+        Inserts a the specified variantSet into this repository.
+        """
+        sql = """
+            INSERT INTO VariantSet (
+                id, datasetId, referenceSetId, name, created, updated,
+                metadata, dataUrlIndexMap)
+            VALUES (?, ?, ?, ?, datetime('now'), datetime('now'), ?, ?);
+        """
+        cursor = self._dbConnection.cursor()
+        # We cheat a little here with the VariantSetMetadata, and encode these
+        # within the table as a JSON dump. These should really be stored in
+        # their own table
+        metadataJson = json.dumps(
+            [metadata.toJsonDict() for metadata in variantSet.getMetadata()])
+        urlMapJson = json.dumps(variantSet.getReferenceToDataUrlIndexMap())
+        cursor.execute(sql, (
+            variantSet.getId(), variantSet.getParentContainer().getId(),
+            variantSet.getReferenceSet().getId(), variantSet.getLocalId(),
+            metadataJson, urlMapJson))
+        for callSet in variantSet.getCallSets():
+            self.insertCallSet(callSet)
+
+    def _readVariantSetTable(self, cursor):
+        cursor.row_factory = sqlite3.Row
+        cursor.execute("SELECT * FROM VariantSet;")
+        for row in cursor:
+            dataset = self.getDataset(row[b'datasetId'])
+            referenceSet = self.getReferenceSet(row[b'referenceSetId'])
```

Comment:
Perfect, solves this long standing issue https://github.com/ga4gh/server/issues/689.